pinot-common/src/main/java/org/apache/pinot/common/metrics/AbstractMetrics.java

```diff
@@ -796,6 +809,7 @@ private void removeGaugeFromMetricRegistry(String metricName) {
   protected abstract G[] getGauges();
 
   protected String getTableName(String tableName) {
-    return _isTableLevelMetricsEnabled || _allowedTables.contains(tableName) ? tableName : "allTables";
+    tableName = _isTableLevelMetricsEnabled || _allowedTables.contains(tableName) ? tableName : "allTables";
+    return StringUtils.replaceOnce(tableName, ".", "#");
```

Comment:
can we please add some comments to explain why we need this change? this will make the code easy to read and understand

Reply:
Sure, will do